src/codegen/function_body_chunk.rs

```diff
@@ -128,12 +120,100 @@ impl Builder {
         let unsafe_ = Chunk::Unsafe(body);
 
         let mut chunks = Vec::new();
+
+        let name = "cancellable".to_string();
+        if self.cancellable {
+            let chunk = Chunk::Let {
+                name: name.clone(),
+                is_mut: false,
+                value: Box::new(Chunk::New { name: "Cancellable".to_string() }),
+                type_: None,
+            };
+            chunks.push(chunk);
+        }
         self.add_into_conversion(&mut chunks);
         self.add_in_array_lengths(&mut chunks);
         self.add_assertion(&mut chunks);
+
+        if let Some(ref trampoline) = self.async_trampoline {
+            chunks.push(Chunk::BoxFn {
+                typ: trampoline.callback_type.clone(),
+            });
+
+            let object_arg =
+                if trampoline.is_method {
+                    "_source_object as *mut _, "
+                } else {
+                    ""
+                };
+            let output_vars = trampoline.output_params.iter()
+                .filter(|param| param.direction == ParameterDirection::Out && param.name != "error")
+                .map(|param| (param, type_mem_mode(env, param)))
+                .map(|(param, mode)| format!("let mut {} = {};\n", param.name, mode_to_string(mode)))
+                .collect::<Vec<_>>()
+                .join("                ");
```

Comment:
Please change it to "\t\t\t\t"

Reply:
I'll instead change that to use `Chunk` directly, to avoid having code as strings.